certbot/storage.py

```diff
@@ -198,6 +209,70 @@ def lineagename_for_filename(config_filename):
             "renewal config file name must end in .conf")
     return os.path.basename(config_filename[:-len(".conf")])
 
+def renewal_filename_for_lineagename(config, lineagename):
+    """Returns the lineagename for a configuration filename.
+    """
+    return os.path.join(config.renewal_configs_dir, lineagename) + ".conf"
+
+def _full_archive_path(config_obj, cli_config, lineagename):
+    """Returns the full archive path for a lineagename"""
+    if "archive_dir" in config_obj:
+        return config_obj["archive_dir"]
+    else:
+        return os.path.join(cli_config.default_archive_dir, lineagename)
+
+def delete_files(config, certname):
+    """Delete all files related to the certificate.
+
+    If some files are not found, ignore them and continue.
+    """
+    renewal_filename = renewal_file_for_certname(config, certname)
+    try:
+        renewal_config = configobj.ConfigObj(renewal_filename)
+    except configobj.ConfigObjError:
+        raise errors.CertStorageError(
```

Comment:
If the renewal file is corrupt, we'd have the option of assuming the user has not renamed the lineage and also renamed another lineage to the same name,  and trying to delete live and archive directories at the names we expect them to have.  That would be the right thing to do 99% of the time.

Alternatively, we could print a log message suggesting that the user might want to consider deleting those directories by hand.